simulators/portal-interop/src/main.rs

```diff
@@ -518,6 +530,77 @@ dyn_async! {
     }
 }
 
+dyn_async! {
+    // test that a node will return a block body via RECURSIVEFINDCONTENT over uTP that it has stored locally
+    async fn test_recursive_find_content_block_body_over_utp<'a> (client_a: Client, client_b: Client) {
+        let block_body_key: HistoryContentKey = serde_json::from_value(json!(BLOCK_BODY_KEY)).unwrap();
+        let block_body_value: HistoryContentValue = serde_json::from_value(json!(BLOCK_BODY_VALUE)).unwrap();
+
+        let header_with_proof_key: HistoryContentKey = serde_json::from_value(json!(HEADER_WITH_PROOF_KEY)).unwrap();
+        let header_with_proof_value: HistoryContentValue = serde_json::from_value(json!(HEADER_WITH_PROOF_VALUE)).unwrap();
+
+        match client_b.rpc.store(block_body_key.clone(), block_body_value.clone()).await {
+            Ok(result) => if !result {
+                panic!("Error storing block body for recursive find content block body over utp");
+            },
+            Err(err) => {
+                panic!("Error storing block body: {err:?}");
+            }
+        }
+
+        match client_b.rpc.store(header_with_proof_key.clone(), header_with_proof_value.clone()).await {
+            Ok(result) => if !result {
+                panic!("Unable to store header with proof for recursive find content");
+            },
+            Err(err) => {
+                panic!("Error storing header with proof for recursive find content: {err:?}");
+            }
+        }
+
+        let target_enr = match client_b.rpc.node_info().await {
+            Ok(node_info) => node_info.enr,
+            Err(err) => {
+                panic!("Error getting node info: {err:?}");
+            }
+        };
+
+        match HistoryNetworkApiClient::add_enr(&client_a.rpc, target_enr.clone()).await {
```

Comment:
Do we need to manually add the enr if `ping` is doing this for us implicitly?

Reply:
resolved